### PR TITLE
Mark explicit template specializations inline

### PR DESCRIPTION
**Pull request by r-devulap**
@Flamefire please review if you can :) This alone won't fix the NumPy build, I will have a patch for that soon.

### CODE REVIEW

File: src/xss-common-includes.h [hunk 1]
```diff
@@ -47,14 +48,17 @@
  * Force inline in cygwin to work around a compiler bug. See
  * https://github.com/numpy/numpy/pull/22315#issuecomment-1267757584
  */
+#define X86_SIMD_SORT_INLINE_ONLY inline
```
Comment (Flamefire) [inline src/xss-common-includes.h:51]: This makes the comment above misplaced as it applies to the define below

**Review by Flamefire — COMMENTED** [review]
As this includes the `inline` for the specializations similar to what [I already tested](https://github.com/intel/x86-simd-sort/issues/111#issuecomment-1851553552) this works.

As for the implementation:
- Why use `X86_SIMD_SORT_INLINE_ONLY` instead of `inline`? Doesn't it make the code less readable?
- And as mentioned in the numpy issue: Why keep the templates at all? Overloads are simpler and would allow using the `X86_SIMD_SORT_INLINE` define (or IMO preferably simply `static`) to create separate instances per TU allowing the Numpy multiple-compilation mode without ODR issues.

**Comment by r-devulap** [comment]
> And as mentioned in the numpy issue: Why keep the templates at all?

Without templates, won't I would need 9 identical copies of all the functions? (qsort, qselect, partialsort, argsort, argselect, etc.) Ex: https://github.com/intel/x86-simd-sort/blob/7060e3c768992441aa6454a6f9320a9fe1f870da/src/xss-common-qsort.h#L554.

File: src/xss-common-includes.h (hunk 1 above)
Comment (r-devulap) [reply src/xss-common-includes.h:51]: Ah, good point. Will fix the order :)

**Comment by Flamefire** [comment]
> I think the macro is pretty self-explanatory. Having a macro gives the ability to make modifications to it easily in one place, if ever needed in the future.

It is much longer and e.g. `X86_SIMD_SORT_INLINE` could be confused with that. Given the name "INLINE_ONLY" what modification would be possibly without changing the name?   
Anyway I have to admit that I dislike macros and try to avoid them whenever possible. Here it seems very odd to have it defined empty. So I'm wondering if that is even correct but I don't see which compilers would actually use that code path.

> > And as mentioned in the numpy issue: Why keep the templates at all?
> 
> Without templates, won't I would need 9 identical copies of all the functions? (qsort, qselect, partialsort, argsort, argselect, etc.) 

True yes, there it is well suited. I was more referring to functions which basically only have specializations and no base implementation. This seems to have changed. Previously this was related to e.g. `avx512_qselect` but now I don't even see the base template anymore (which was only declared but not defined). Only the specialization in src/avx512fp16-16bit-qsort.hpp. How does that work? What am I missing?

Anyway that comment doesn't seem to apply anymore as I missed what was changed in other PRs/commits such as 472c7d0

**Pull request merged**